app/src/androidTest/java/com/github/sdp_begreen/begreen/fragments/FollowersFragmentTest.kt

```diff
@@ -1,41 +1,102 @@
 package com.github.sdp_begreen.begreen.fragments
 
 import android.os.Bundle
-import androidx.fragment.app.testing.launchFragmentInContainer
-import androidx.test.espresso.Espresso
-import androidx.test.espresso.assertion.ViewAssertions
-import androidx.test.espresso.matcher.ViewMatchers
+import androidx.fragment.app.testing.FragmentScenario
+import androidx.recyclerview.widget.RecyclerView
+import androidx.test.ext.junit.rules.ActivityScenarioRule
 import androidx.test.ext.junit.runners.AndroidJUnit4
 import androidx.test.filters.LargeTest
 import com.github.sdp_begreen.begreen.R
+import com.github.sdp_begreen.begreen.activities.MainActivity
+import com.github.sdp_begreen.begreen.fragments.FollowersFragment.Companion.ARG_COLUMN_COUNT
+import com.github.sdp_begreen.begreen.fragments.FollowersFragment.Companion.ARG_USER_LIST
+import com.github.sdp_begreen.begreen.models.User
+import com.github.sdp_begreen.begreen.rules.KoinTestRule
+import junit.framework.TestCase
+import org.hamcrest.CoreMatchers
+import org.hamcrest.CoreMatchers.equalTo
+import org.hamcrest.MatcherAssert
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.Before
+import org.junit.Rule
 import org.junit.Test
 import org.junit.runner.RunWith
 
 @RunWith(AndroidJUnit4::class)
 @LargeTest
 class FollowersFragmentTest {
+
+    @get:Rule
+    val activityRule = ActivityScenarioRule(MainActivity::class.java)
+
+    @get:Rule
+    val koinTestRule = KoinTestRule()
+
+    private lateinit var followers: ArrayList<User>
+
+    @Before
+    fun setup() {
+        // Initialize test data
+        followers = arrayListOf(
+            User("id1", 1, "User1"),
+            User("id2", 2, "User2"),
+            User("id3", 3, "User3"),
+            User("id2", 4, "User4"),
+            User("id2", 5, "User5")
+        )
+    }
+
     @Test
-    fun testAdviceFragment() {
-        launchFragmentInContainer<FollowersFragment>()
-        Espresso.onView(ViewMatchers.withId(R.id.followersFragmentTextView))
-            .check(ViewAssertions.matches(ViewMatchers.withText("Fragment where we can display the user's followers")))
+    fun onCreateViewWithValidViewReturnsView() {
+        // Set up fragment arguments
+        val args = Bundle().apply {
+            putInt(ARG_COLUMN_COUNT, 1)
+            putParcelableArrayList(ARG_USER_LIST, followers)
+        }
+
+        // Launch fragment with arguments
+        val scenario = FragmentScenario.launchInContainer(UserFragment::class.java, args)
+        // Wait for the fragment to be created
+        scenario.onFragment { fragment ->
+            val recyclerView = fragment.view
+            TestCase.assertNotNull(recyclerView)
+            TestCase.assertTrue(recyclerView is RecyclerView)
```

Comment:
I think you could use the Android Assert class here instead of TestCase: Assert.assertNotNull(recyclerView)
It seems that assert is more consistent with Android tests. What do you think ?

Reply:
Yes indeed, I used `assertThat` to be more consistent with the other tests, let me know what you think about it !